server/datastore/mysql/migrations/tables/20210721171531_AddCPETable.go

```diff
@@ -0,0 +1,34 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20210721171531, Down_20210721171531)
+}
+
+func Up_20210721171531(tx *sql.Tx) error {
+	sql := `
+		CREATE TABLE IF NOT EXISTS software_cpe (
+			id int(10) unsigned NOT NULL AUTO_INCREMENT,
+			software_id bigint unsigned,
+			created_at timestamp DEFAULT CURRENT_TIMESTAMP,
+			updated_at timestamp NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+			cpe varchar(255) NOT NULL,
+			PRIMARY KEY (id),
+			FOREIGN KEY (software_id) REFERENCES software(id) ON DELETE CASCADE
+		)
```

Comment:
I don't think so, but I'm not sure. We could have multiple "candidate" CPEs, even if technically there's only one.

If we add this to the `host_software` table we would also have to add the timestamps for future update/checks logic.

The other concern is table locking: the `host_software` table is going to be updated/deleted in parallel to this, and given the previous performance concerns, this will be the most "generic" approach in terms of engine and locking. It feels a bit over cautions depending on at what point of the day you catch me though :)

Reply:
Sounds good, I think we can leave it as-is.